test/lit/passes/optimize-instructions.wast

```diff
@@ -12013,4 +12024,108 @@
       )
     )
   )
+  ;; CHECK:      (func $ternary-identical-arms-br_if-same (param $x i32) (param $y i32) (param $z i32)
+  ;; CHECK-NEXT:  (block $block
+  ;; CHECK-NEXT:   (br_if $block
+  ;; CHECK-NEXT:    (if (result i32)
+  ;; CHECK-NEXT:     (local.get $z)
+  ;; CHECK-NEXT:     (local.get $x)
+  ;; CHECK-NEXT:     (local.get $y)
```

Comment:
It looks like we could further simplify `if`s into `select`s whenever the only side effects are reading.

Reply:
I believe we already do that. A read is not a side effect, and RemoveUnusedBrs has a `selectify` method to turn relevant ifs into selects. It has some heuristics based on how expensive executing both sides,

https://github.com/WebAssembly/binaryen/blob/cb776da297a845e15171405b7f7518619122d7aa/src/passes/RemoveUnusedBrs.cpp#L1057-L1063